hammers/scripts/maintenance_reservation.py

```diff
@@ -1,144 +1,224 @@
+    end_time_str_in_ct = end_time.replace(tzinfo=tz.gettz('UTC')).astimezone(tz.gettz('America/Chicago')).strftime(DATETIME_STR_FORMAT)
+    
+    print("Creating maintenance reservation for node {node_name} (id: {node_uuid}), starting {start} and ending {end} in central time".format(node_name=node.name, 
+                                                                                                                                              node_uuid=node.uuid,
+                                                                                                                                              start=start_time_str_in_ct,
+                                                                                                                                              end=end_time_str_in_ct))
+    if not dryrun:
+        blazar = blazar_client.Client(1, session=sess, service_type='reservation', region_name=region)
+        resource_properties = '["=", "$uid", "{node_uuid}"]'.format(node_uuid=node.uuid)
+        phys_res = {'min': "1", 'max': "1", 'hypervisor_properties': "", 'resource_properties': resource_properties, 'resource_type': 'physical:host'}
+        lease = blazar.lease.create(name=MAINT_LEASE_NAME.format(node_name=node.name.replace(' ', '_'), operator=operator.replace(' ', '_'), reason=reason.replace(' ', '_')),
+                                    start=start_time.strftime('%Y-%m-%d %H:%M'),
+                                    end=end_time.strftime('%Y-%m-%d %H:%M'),
+                                    reservations=[phys_res],
+                                    events=[])
+        print("Lease {name} (id: {id}) created successfully!".format(name=lease['name'], id=lease['id']))
+    
+    return start_time_str_in_ct, end_time_str_in_ct
 
+@prometheus_exporter(__file__)
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv
+
+    parser = argparse.ArgumentParser(description='Reserve nodes for maintenance')
+    append_global_identity_args(parser, argv)
+    
+    mysqlargs = MySqlArgs({
+        'user': 'root',
+        'password': '',
+        'host': 'localhost',
+        'port': 3306,
+    })
+    mysqlargs.inject(parser)
+    
+    parser.add_argument('--operator', type=str, required=True, help='Chameleon account username of the operator')
+    parser.add_argument('--nodes', type=str, required=True, help='node ids or node names; comma separated')
+    parser.add_argument('--reason', type=str, required=True, help='maintenance reasons')
+    parser.add_argument('--dry-run', action="store_true", help='perform a trial run without making reservations')
+    parser.add_argument('--start-time', type=valid_date, default=None, help='lease start time in CT (YYYY-mm-DD HH:MM:SS); if not given, start at the earliest possible datetime')
+    parser.add_argument('--estimate-hours', type=int, default=24, help='estimated hours required for maintenance; default is 24 hours')
```

Comment:
I recommend increasing this to at least a week. Typically nodes in maintenance can need a lot of time to get to.